src/iota/ed25519.c

```diff
@@ -27,24 +27,36 @@ uint8_t ed25519_get_key_pair(uint32_t *bip32_path, uint32_t bip32_path_length,
                              cx_ecfp_public_key_t *pub)
 {
     uint8_t keySeed[32];
+    uint8_t ret = 1;
 
     // getting the seed to derive and configuring it with SLIP10
     os_perso_derive_node_bip32_seed_key(
         HDW_ED25519_SLIP10, CX_CURVE_Ed25519, bip32_path, bip32_path_length,
         keySeed, NULL, (unsigned char *)"ed25519 seed", 12);
 
-    // initializing the private key and public key instance
-    // with selected curve ED25519
-    cx_ecfp_init_private_key(CX_CURVE_Ed25519, keySeed, sizeof(keySeed), pk);
-    cx_ecfp_init_public_key(CX_CURVE_Ed25519, NULL, 0, pub);
-
-    // generating the key pair
-    cx_ecfp_generate_pair(CX_CURVE_Ed25519, pub, pk, 1);
-
-    // resetting the variables to avoid leak
-    explicit_bzero(keySeed, sizeof(keySeed));
```

Comment:
if an exception is thrown by cx_lib the `keySeed` would not be wiped from memory